src/checkpoint/checkpoint_txn.c

```diff
@@ -1740,6 +1701,7 @@ __checkpoint_db_internal(WT_SESSION_IMPL *session, const char *cfg[])
     __wt_free(session, session->ckpt.handle);
     WT_ASSERT(session, session->ckpt.crash_point == 0);
     session->ckpt.handle_allocated = session->ckpt.handle_next = session->ckpt.crash_point = 0;
+    session->ckpt.key_provider_crash_point = -1;
```

Comment:
Here is the place where special value `CRASH_POINT_NONE` would be helpful and make code more readable.